### PR TITLE
Api proxy

### PR DESCRIPTION
**Pull request by jonsherrard**
First draft at API proxy

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,8 +28,8 @@
     "eslint-plugin-react": "^6.8.0",
     "fetch": "^1.1.0",
     "glamor": "^2.20.12",
-    "h2o2": "^5.4.0",
-    "hapi": "^16.0.1",
+    "h2o2": "5.1.1",
+    "hapi": "15.0.3",
```
Comment (jonsherrard) [inline package.json]: These can go back up. Downgraded in debugging

**Comment by jonsherrard** [comment]
[Create API proxy](https://trello.com/c/JiIzcv5R/11-create-api-proxy)

File: src/loader-home.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ export default class Loader extends Component {
     const customLoader = loadContext.loaders.Home
     if (customLoader) return customLoader(loadContext.siteUrl, cb)
 
-    const baseUrl = `${loadContext.siteUrl}/wp-json/wp/v2`
+    const baseUrl = `${loadContext.serverUri || window.location.origin}/api/v1`
```
Comment (ecoad) [inline src/loader-home.js]: `location.origin` isn't compatible with IE <11
https://developer.mozilla.org/en-US/docs/Web/API/Window/location

File: src/server.js [hunk 3]
```diff
@@ -52,13 +54,28 @@ export default class TapestryServer {
     })
   }
 
+  routeApi () {
```
Comment (ecoad) [inline src/server.js]: It's crazy how little code is needed to achieve this feature. Nice

File: src/loader-home.js (hunk 2 above)
Comment (jonsherrard) [inline src/loader-home.js]: Good catch:

`if (!window.location.origin) {
  window.location.origin = window.location.protocol + "//" + window.location.hostname + (window.location.port ? ':' + window.location.port: '');
}`
Will need to implement this

File: src/server.js (hunk 3 above)
Comment (jonsherrard) [inline src/server.js]: It's a basic, and slow implementation, I think we'll write a `proxyHandler` method that does more involved logic, like rewriting headers, and maybe a bit of Tapestry-side caching.

**Pull request merged**